library/ssl_misc.h

```diff
@@ -751,9 +751,11 @@ struct mbedtls_ssl_handshake_params {
     mbedtls_dhm_context dhm_ctx;                /*!<  DHM key exchange        */
 #endif
 
-#if defined(MBEDTLS_ECDH_C) && !defined(MBEDTLS_USE_PSA_CRYPTO)
+#if !defined(MBEDTLS_USE_PSA_CRYPTO) && \
+    defined(MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_1_2_ENABLED)
     mbedtls_ecdh_context ecdh_ctx;              /*!<  ECDH key exchange       */
-#endif /* MBEDTLS_ECDH_C && !MBEDTLS_USE_PSA_CRYPTO */
+#endif /* !MBEDTLS_USE_PSA_CRYPTO &&
+          MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_1_2_ENABLED */
 
 #if defined(PSA_WANT_ALG_ECDH) && defined(PSA_WANT_ALG_FFDH)
```

Comment:
@mpg I know that there are still these dependencies on `PSA_WANT_ALG_ECDH` and `PSA_WANT_ALG_FFDH` here that are not in the "key exchange" format that we would like to achieve in this PR. However I had some doubt if replacing also these symbols with the key exchange ones could make the code easier to read or more complex. Wdyt?

Reply:
I mean, the best that I could think of was a list of the following guards for defining the same stuff:
```
#if (defined(MBEDTLS_KEY_EXCHANGE_SOME_ECDH_OR_ECDHE_1_2_ENABLED) && \
    defined(MBEDTLS_KEY_EXCHANGE_SOME_DHE_ENABLED) && \
    defined(MBEDTLS_USE_PSA_CRYTO)) || \
    (defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_ECDHE_ENABLED) && \
    defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_FFDH_ENABLED))
```

But I have a question for this: is `ssl_misc` the right place for this definition? I mean both `MBEDTLS_PSA_MAX_FFDH_PUBKEY_LENGTH` and `MBEDTLS_PSA_MAX_EC_PUBKEY_LENGTH` are defined in `psa_util.h` so perhaps that is the right place to put this. Wdyt?

Reply:
Right, `psa_util.h` looks like the right place to define what we currently call `SSL_XXDH_PSA_PEERKEY_SIZE`, except then it should have a more generic name, like `MBEDTLS_PSA_MAX_KEY_AGREEMENT_PUBKEY_LENGTH`.

Reply:
There's `PSA_EXPORT_PUBLIC_KEY_MAX_SIZE` which is PSA standard, is it good enough here?

Mbed TLS also defines `PSA_KEY_EXPORT_ECC_PUBLIC_KEY_MAX_SIZE(PSA_VENDOR_ECC_MAX_CURVE_BITS)` and `PSA_KEY_EXPORT_FFDH_PUBLIC_KEY_MAX_SIZE(PSA_VENDOR_FFDH_MAX_KEY_BITS)`.

Reply:
Right, it seems that `PSA_EXPORT_PUBLIC_KEY_MAX_SIZE` might be a good alternative instead of defining a new symbol here.
The only drawback I see is that by using `PSA_EXPORT_PUBLIC_KEY_MAX_SIZE` we are always allocating buffer for 8192 bits (`PSA_VENDOR_FFDH_MAX_KEY_BITS`) even if `PSA_WANT_ALG_FFDH` is not defined, so we might end up using more memory than what it is strictly required. Isn't that a problem?

Reply:
Seems good! The thought had crossed my mind that perhaps there was already a PSA macro for that, but then I forgot to check :/